maas-api/internal/token/handler.go

```diff
@@ -29,40 +31,82 @@ func NewHandler(name string, manager TokenManager) *Handler {
 	}
 }
 
-// ExtractUserInfo validates kubernetes tokens.
-func (h *Handler) ExtractUserInfo(reviewer *Reviewer) gin.HandlerFunc {
+// parseGroupsHeader parses the X-MAAS-GROUP header which comes as a JSON array.
+// Format: "[\"group1\",\"group2\",\"group3\"]" (JSON-encoded array string).
+func parseGroupsHeader(header string) ([]string, error) {
+	if header == "" {
+		return nil, errors.New("header is empty")
+	}
+
+	// Try to unmarshal as JSON array directly
+	var groups []string
+	if err := json.Unmarshal([]byte(header), &groups); err != nil {
+		return nil, fmt.Errorf("failed to parse header as JSON array: %w", err)
+	}
+
+	if len(groups) == 0 {
+		return nil, errors.New("no groups found in header")
+	}
+
+	// Trim whitespace from each group
+	for i := range groups {
+		groups[i] = strings.TrimSpace(groups[i])
+	}
+
+	return groups, nil
+}
+
+// ExtractUserInfo extracts user information from X-MAAS-* headers set by the auth policy.
+func (h *Handler) ExtractUserInfo() gin.HandlerFunc {
 	return func(c *gin.Context) {
-		authHeader := c.GetHeader("Authorization")
-		if authHeader == "" {
-			c.JSON(http.StatusUnauthorized, gin.H{"error": "Authorization header required"})
+		username := strings.TrimSpace(c.GetHeader("X-MAAS-USERNAME"))
+		groupHeader := c.GetHeader("X-MAAS-GROUP")
+		source := strings.TrimSpace(c.GetHeader("X-MAAS-SOURCE"))
+
+		// Validate required headers exist and are not empty
+		if username == "" {
+			log.Printf("Missing or empty X-MAAS-USERNAME header")
+			c.JSON(http.StatusUnauthorized, gin.H{"error": "X-MAAS-USERNAME header required and must not be empty"})
 			c.Abort()
 			return
 		}
 
-		if !strings.HasPrefix(authHeader, "Bearer ") {
-			c.JSON(http.StatusUnauthorized, gin.H{"error": "Invalid authorization format. Use: Authorization: Bearer <token>"})
+		if groupHeader == "" {
+			log.Printf("Missing X-MAAS-GROUP header for user: %s", username)
+			c.JSON(http.StatusUnauthorized, gin.H{"error": "X-MAAS-GROUP header required"})
 			c.Abort()
 			return
 		}
 
-		token := strings.TrimPrefix(authHeader, "Bearer ")
-
-		// Validate with Manager (K8s validation)
-		userContext, err := h.manager.ValidateToken(c.Request.Context(), token, reviewer)
```

Comment:
`ValidateToken` is dead code now, and also makes the whole `Reviewer` obsolete as this PR moves the logic of user extraction up to AuthPolicy level - I think we can remove both.